core/logic_events.go

```diff
@@ -166,6 +166,22 @@ func (e eventsLogic) processWebToolsEvents() {
 		return
 	}
 
+	// Keep only one instance per unique EventID to prevent duplicates in the app.
```

Comment:
Define a function with aprropriate name which makes this. Keep the code well structured.